webapp/api/launchpad.py

```diff
@@ -1,6 +1,105 @@
-from launchpadlib.launchpad import Launchpad
+from hashlib import md5
+from webapp.api.requests import Session
 
 
-launchpad = Launchpad.login_anonymously(
-    "storefront", "production", version="devel"
-)
+class Launchpad:
+    """
+    Provides authentication for Launchpad.
+    """
+
+    BASE_URL = "https://api.launchpad.net"
+
+    def __init__(self, username, token, signature, session=Session()):
+        self.username = username
+        self.session = session
+        self.session.headers["Accept"] = "application/json"
+        self.session.headers["Authorization"] = (
+            f'OAuth oauth_version="1.0", '
+            f'oauth_signature_method="PLAINTEXT", '
+            f"oauth_consumer_key={username}, "
+            f'oauth_token="{token}", '
+            f'oauth_signature="&{signature}"'
+        )
+
+    def _request(self, method, url, params={}, data={}):
+        """
+        Makes a raw HTTP request and returns the response.
+        """
+        if not url.startswith("http"):
+            url = f"{self.BASE_URL}/devel/{url}"
+
+        response = self.session.request(method, url, params=params, data=data)
+        response.raise_for_status()
+
+        return response
+
+    def get(self, resource, params=None):
+        """
+        Makes a raw HTTP GET request and returns the response.
+        """
+        response = self._request("GET", resource, params=params)
+
+        return response.json()
```

Comment:
Should we be doing a [`response.raise_for_status()`](https://2.python-requests.org/en/master/api/#requests.Response.raise_for_status) call in all these functions? So that we get clearer error messages when we get erroneous responses from the API.

Reply:
Hey @nottrobin , but we are, right? The `_request` method is doing it

Reply:
Oh I see! Wow. In that case, I will change my critique to:

These `get` and `post` methods seem really redundant since we already have a `_request` method...

Maybe the `_request` method could be improved a bit by defaulting `method` to `"GET"`? I'm also suggesting a similar thing in @carkod's PR: https://github.com/canonical-web-and-design/anbox-cloud.io/pull/45#discussion_r370720385

Reply:
Okay, I thought the `get` and `post` methods will be useful.